cmd/fs-v1-helpers.go

```diff
@@ -309,14 +309,19 @@ func fsCreateFile(ctx context.Context, filePath string, reader io.Reader, falloc
 	if err != nil {
 		return 0, osErrToFileErr(err)
 	}
-	defer writer.Close()
 
 	bytesWritten, err := io.Copy(writer, reader)
 	if err != nil {
 		logger.LogIf(ctx, err)
 		return 0, err
 	}
 
+	err = writer.Close()
+	if err != nil {
+		logger.LogIf(ctx, err)
+		return 0, err
+	}
```

Comment:
I suggest replacing the existing code with

```golang
defer func() {
        err = writer.Close()
	if err != nil {
		logger.LogIf(ctx, err)
		return 0, err
	}
}()
```

This will act like a `finally`block and ensure that the writer is always closed, even when a write error occurs.